src/main/java/edu/cornell/kfs/concur/businessobjects/ValidationResult.java

```diff
@@ -43,15 +44,25 @@ public void addMessage(String message) {
         if (messages == null) {
             messages = new ArrayList<String>();
         }
-        messages.add(message);
+        if (StringUtils.isNotBlank(message) && isNotDuplicateMessage(message)) {
+            messages.add(message);
+        }
+    }
+    
+    private boolean isNotDuplicateMessage(String message) {
+        for (String currentMessage : getMessages()) {
+            if (StringUtils.equals(currentMessage, message)) {
```

Comment:
I do not believe that this will deal with string case of the message contents..... meaning I don't think it will see "Error Encountered" as a duplicate of "ERROR encountered".  Is that an issue?